services/frontend-service/pkg/interceptors/interceptors.go

```diff
@@ -124,6 +125,22 @@ func GoogleIAPInterceptor(
 	httpHandler(w, req)
 }
 
+func CheckPolicy(httpCtx context.Context, w http.ResponseWriter, req *http.Request, userGroup, DexRbacPolicyPath string) context.Context {
+	policy, err := auth.ReadRbacPolicy(true, DexRbacPolicyPath)
+	if err != nil {
+		http.Error(w, "unable to access RBAC policy to validate login", http.StatusBadRequest)
+		return nil
+	}
+	for _, policyGroup := range policy.Groups {
+		if policyGroup.Group == userGroup {
+			auth.WriteUserRoleToHttpHeader(req, policyGroup.Role)
+			httpCtx = auth.WriteUserRoleToGrpcContext(req.Context(), policyGroup.Role)
+			httpCtx = context.WithValue(httpCtx, auth.HeaderUserRole, auth.Encode64(policyGroup.Role)) //nolint: go-staticcheck
```

Comment:
this is a bit hard to follow.
Do we really need to write this role into 3 places? Seems a bit excessive.
Please check if this is really necessary - and if it is, please add a comment, e.g
`auth.WriteUserRoleToHttpHeader(req, policyGroup.Role) // this is needed when service foo calls service bla over http`
etc.